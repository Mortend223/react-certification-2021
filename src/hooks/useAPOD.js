import { useMemo } from 'react';
import useFetchNASAData from './useFetchNasaData';

function parseResponse(response) {
  return response.data;
}

function useAPOD(query) {
  const params = useMemo(() => {
    return {
      api_key: process.env.REACT_APP_NASA_API_KEY,
      date: query,
    };
  }, [query]);
  const { response, error, loading } = useFetchNASAData(params, parseResponse);
  return { response: response ?? null, error, loading };
}

export default useAPOD;
