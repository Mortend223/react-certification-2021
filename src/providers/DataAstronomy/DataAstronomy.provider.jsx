import React, { createContext, useContext, useEffect, useReducer } from 'react';
import PropTypes from 'prop-types';

const DataAstromyContext = createContext(null);

function DataAstronomyReducer(state, action) {
  switch (action.type) {
    case 'updateDate': {
      return { ...state, searchDate: action.date };
    }
    default: {
      throw new Error(`Unhandled action type: ${action.type}`);
    }
  }
}

function DataAstronomyProvider({ children }) {
  const [state, dispatch] = useReducer(DataAstronomyReducer, {
    searchDate: new Date(),
  });

  const onChangeDate = (value) => {
    dispatch({ type: 'updateDate', date: value });
  };

  useEffect(() => {}, [state.searchDate]);

  return (
    <DataAstromyContext.Provider
      value={{
        searchDate: state.searchDate,
        onChangeDate,
      }}
    >
      {children}
    </DataAstromyContext.Provider>
  );
}

function useDataAstronomy() {
  const context = useContext(DataAstromyContext);
  if (context === undefined) {
    throw new Error('useDataAstronomy must be used within a DataAstronomyProvider');
  }
  return context;
}

DataAstronomyProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export { useDataAstronomy };
export default DataAstronomyProvider;
