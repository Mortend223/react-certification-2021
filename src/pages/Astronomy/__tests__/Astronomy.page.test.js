import React from 'react';
import { render } from '@testing-library/react';

// Components
import AstronomyPage from '../Astronomy.page';

// Hooks
import useAPOD from '../../../hooks/useAPOD';
import { useDataAstronomy } from '../../../providers/DataAstronomy/DataAstronomy.provider';

jest.mock('../../../hooks/useAPOD');
jest.mock('../../../providers/DataAstronomy/DataAstronomy.provider');

describe('Astronomy Page', () => {
  it('Should show Astronomy Page', () => {
    useDataAstronomy.mockReturnValue({
      searchDate: new Date(),
    });
    useAPOD.mockReturnValue({
      response: {
        explanation: "We've seen this same supernova three times",
        title: 'SN Requiem: A Supernova Seen Three Times So Far',
        url: 'https://apod.nasa.gov/apod/image/2111/MACSJ0138_Hubble_1080.jpg',
      },
      error: {
        code: 400,
        msg: 'Some text',
      },
      loading: false,
    });
    const container = render(<AstronomyPage />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
