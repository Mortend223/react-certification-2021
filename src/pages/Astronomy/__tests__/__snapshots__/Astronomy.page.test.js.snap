// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Astronomy Page Should show Astronomy Page 1`] = `undefined`;
